### PR TITLE
Update pom.xml

### PR DESCRIPTION
**Pull request by harvies**
Support jdk1.9

closes https://github.com/apache/skywalking/issues/5501

### CODE REVIEW

**Comment by kezhenxu94** [comment]
It's just a test tool, why bother to add this? Do you build this tool locally and only have JDK9 installed?

**Review by kezhenxu94 — APPROVED** [review]

**Comment by harvies** [comment]
> It's just a test tool, why bother to add this? Do you build this tool locally and only have JDK9 installed?

There are multiple jdk installed locally, the default is jdk11.

**Comment by wu-sheng** [comment]
Why merge this? We are asking for jdk8 compiling im the doc. And as a test tool, we don't need multiple JDK supports.

**Comment by wu-sheng** [comment]
If this PR is required, the document should be updated to tell users, you need a `-p` in the maven command.

File: mock-collector/pom.xml [hunk 1]
```diff
@@ -187,4 +187,19 @@
             </plugin>
         </plugins>
     </build>
+    <profiles>
+        <profile>
+            <id>java9</id>
+            <activation>
+                <jdk>[1.9,]</jdk>
```
Comment (kezhenxu94) [inline mock-collector/pom.xml:194]: This is automatically detected and activated, so it's transparent to the users, they don't need to care about the JDK version

**Comment by wu-sheng** [comment]
@harvies To verify this PR is no harm, could you update the main repo commit id?

File: mock-collector/pom.xml (hunk 1 above)
Comment (harvies) [reply mock-collector/pom.xml:194]: yes

**Comment by harvies** [comment]
> @harvies To verify this PR is no harm, could you update the main repo commit id?

Please see https://github.com/apache/skywalking/pull/5522

**Pull request merged**